src/test/java/seedu/address/model/candidate/SeniorityTest.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.model.candidate;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class SeniorityTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        Integer invalidSeniority = null;
+        assertThrows(NullPointerException.class, () -> new Seniority(invalidSeniority));
+    }
+
+    @Test
+    public void constructor_invalidSeniority_throwsIllegalArgumentException() {
+        int invalidSeniority = 0;
+        assertThrows(IllegalArgumentException.class, () -> new Seniority(invalidSeniority));
+    }
+
```

Comment:
Was thinking we can write test cases here for more edge cases like if seniority is >= 1 and <= 4, but it's a `double` instead of an `int`

Reply:
@tiewweijian thank you for reviewing and proposed suggestions have been added!

just a note, clarified certain things offline too.